client/src/components/forms/submit-button.component.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { describe, expect, it } from 'vitest';
+import { SubmitButton } from './submit-button.component';
+import { render, screen } from '~/test/utils';
+import { GBIcon } from '../../i18n/icons';
+describe('Submit Button', () => {
+  it('should render a submit button with text only', () => {
+    render(<SubmitButton isDisabled>Test text</SubmitButton>);
+    expect(screen.getByRole('button', { name: 'Test text' })).toBeInTheDocument();
+  });
+  it('should render a submit button with an icon only', () => {
+    render(
+      <SubmitButton isDisabled>
+        <GBIcon />
+      </SubmitButton>,
+    );
+    expect(screen.getByRole('button', { name: 'United kingdom Flag' })).toBeInTheDocument();
+  });
```

Comment:
Again, your not using the `isDisabled`, so unless you are going to use it, it should be removed from the render.